### PR TITLE
Add a guard before loading the whitelisted classes

### LINKED ISSUE #336: class_alias breaks if class exists
## Bug report

| Question    | Answer
| ------------| ---------------
| Box version | 3.8
| PHP version | 7.3.5
| MacOs
| Github Repo | - https://github.com/vimeo/psalm

PHP Scoper slightly breaks [this polyfill for `JsonException`](https://github.com/symfony/polyfill-php73/blob/d1fb4abcc0c47be136208ad9d68bf59f1ee17abd/Resources/stubs/JsonException.php).

It gets a `class_alias` appended (as expected), but that `class_alias` causes a warning when running on PHP 7.3 as PHP 7.3 includes the class.

```php
\class_alias('_HumbugBoxc9de1eddd699\\JsonException', 'JsonException', \false);
```

I'm currently fixing it in Psalm's build with [this hideous patch](https://github.com/vimeo/psalm/blob/b13eaa20d279e37bc5083bde749147c3265f40df/bin/improve_class_alias.php) that wraps a `class_exists` check around the `class_alias`:

```php
if (!\class_exists(\JsonException::class, false)) {
    \class_alias('_HumbugBox9de1eddd699\\JsonException', 'JsonException', \false);
}
```

### LINKED ISSUE #338: "Cannot declare class JsonException" when using PHAR of php-scoper 0.12.2
#### Excerpt from output of `ant signed-scoped-phar`
```
PHP Warning:  Cannot declare class JsonException, because the name is already in use in phar:///usr/local/src/phpunit/tools/php-scoper/vendor/symfony/polyfill-php73/Resources/stubs/JsonException.php on line 8
```

#### Complete output of `ant signed-scoped-phar`
```
Buildfile: /usr/local/src/phpunit/build.xml

clean:
   [delete] Deleting directory /usr/local/src/phpunit/vendor
   [delete] Deleting: /usr/local/src/phpunit/composer.lock

-dependencies-installed:

validate-composer-json:
 [composer] /usr/local/src/phpunit/composer.json is valid

install-dependencies:
     [copy] Copying 1 file to /usr/local/src/phpunit
 [composer] ./composer.json has been updated
 [composer] Loading composer repositories with package information
 [composer] Updating dependencies (including require-dev)
 [composer] Package operations: 29 installs, 0 updates, 0 removals
 [composer]   - Installing myclabs/deep-copy (1.9.1): Loading from cache
 [composer]   - Installing phar-io/version (2.0.1): Loading from cache
 [composer]   - Installing phar-io/manifest (1.0.3): Loading from cache
 [composer]   - Installing sebastian/recursion-context (3.0.0): Loading from cache
 [composer]   - Installing doctrine/instantiator (1.2.0): Loading from cache
 [composer]   - Installing sebastian/exporter (3.1.0): Loading from cache
 [composer]   - Installing sebastian/diff (3.0.2): Loading from cache
 [composer]   - Installing sebastian/comparator (3.0.2): Loading from cache
 [composer]   - Installing symfony/polyfill-ctype (v1.11.0): Loading from cache
 [composer]   - Installing webmozart/assert (1.4.0): Loading from cache
 [composer]   - Installing phpdocumentor/reflection-common (1.0.1): Loading from cache
 [composer]   - Installing phpdocumentor/type-resolver (0.4.0): Loading from cache
 [composer]   - Installing phpdocumentor/reflection-docblock (4.3.1): Loading from cache
 [composer]   - Installing phpspec/prophecy (1.8.0): Loading from cache
 [composer]   - Installing theseer/tokenizer (1.1.2): Loading from cache
 [composer]   - Installing sebastian/version (2.0.1): Loading from cache
 [composer]   - Installing sebastian/environment (4.2.2): Loading from cache
 [composer]   - Installing sebastian/code-unit-reverse-lookup (1.0.1): Loading from cache
 [composer]   - Installing phpunit/php-text-template (1.2.1): Loading from cache
 [composer]   - Installing phpunit/php-token-stream (3.0.1): Loading from cache
 [composer]   - Installing phpunit/php-file-iterator (2.0.2): Loading from cache
 [composer]   - Installing phpunit/php-code-coverage (7.0.5): Loading from cache
 [composer]   - Installing phpunit/php-invoker (2.0.0): Loading from cache
 [composer]   - Installing phpunit/php-timer (2.1.2): Loading from cache
 [composer]   - Installing sebastian/object-reflector (1.1.1): Loading from cache
 [composer]   - Installing sebastian/global-state (3.0.0): Loading from cache
 [composer]   - Installing sebastian/object-enumerator (3.0.3): Loading from cache
 [composer]   - Installing sebastian/resource-operations (2.0.1): Loading from cache
 [composer]   - Installing sebastian/type (1.1.1): Loading from cache
 [composer] Writing lock file
 [composer] Generating optimized autoload files
     [move] Moving 1 file to /usr/local/src/phpunit

-phar-prepare:
    [mkdir] Created dir: /usr/local/src/phpunit/build/phar
    [mkdir] Created dir: /usr/local/src/phpunit/build/phar-scoped
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar-scoped
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/php-code-coverage
     [copy] Copying 61 files to /usr/local/src/phpunit/build/phar/php-code-coverage
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/php-file-iterator
     [copy] Copying 3 files to /usr/local/src/phpunit/build/phar/php-file-iterator
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/php-text-template
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/php-text-template
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/php-timer
     [copy] Copying 3 files to /usr/local/src/phpunit/build/phar/php-timer
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/php-token-stream
     [copy] Copying 3 files to /usr/local/src/phpunit/build/phar/php-token-stream
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-code-unit-reverse-lookup
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-code-unit-reverse-lookup
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-comparator
     [copy] Copying 15 files to /usr/local/src/phpunit/build/phar/sebastian-comparator
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-diff
     [copy] Copying 16 files to /usr/local/src/phpunit/build/phar/sebastian-diff
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-environment
     [copy] Copying 3 files to /usr/local/src/phpunit/build/phar/sebastian-environment
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-exporter
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-exporter
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-recursion-context
     [copy] Copying 3 files to /usr/local/src/phpunit/build/phar/sebastian-recursion-context
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-resource-operations
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-resource-operations
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-global-state
     [copy] Copying 6 files to /usr/local/src/phpunit/build/phar/sebastian-global-state
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/object-enumerator
     [copy] Copying 3 files to /usr/local/src/phpunit/build/phar/sebastian-object-enumerator
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/object-reflector
     [copy] Copying 3 files to /usr/local/src/phpunit/build/phar/sebastian-object-reflector
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-type
     [copy] Copying 11 files to /usr/local/src/phpunit/build/phar/sebastian-type
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-version
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/sebastian-version
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/doctrine-instantiator
     [copy] Copying 5 files to /usr/local/src/phpunit/build/phar/doctrine-instantiator
     [copy] Copying 3 files to /usr/local/src/phpunit/build/phar/php-invoker
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/phpdocumentor-reflection-common
     [copy] Copying 6 files to /usr/local/src/phpunit/build/phar/phpdocumentor-reflection-common
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/phpdocumentor-reflection-docblock
     [copy] Copying 38 files to /usr/local/src/phpunit/build/phar/phpdocumentor-reflection-docblock
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/phpdocumentor-type-resolver
     [copy] Copying 24 files to /usr/local/src/phpunit/build/phar/phpdocumentor-type-resolver
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/phpspec-prophecy
     [copy] Copying 89 files to /usr/local/src/phpunit/build/phar/phpspec-prophecy
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/myclabs-deep-copy
     [copy] Copying 24 files to /usr/local/src/phpunit/build/phar/myclabs-deep-copy
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/webmozart-assert
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/webmozart-assert
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/phar-io-manifest
     [copy] Copying 49 files to /usr/local/src/phpunit/build/phar/phar-io-manifest
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/phar-io-version
     [copy] Copying 18 files to /usr/local/src/phpunit/build/phar/phar-io-version
     [copy] Copying 1 file to /usr/local/src/phpunit/build/phar/theseer-tokenizer
     [copy] Copying 8 files to /usr/local/src/phpunit/build/phar/theseer-tokenizer

-phar-determine-version:

scoped-phar:

-phar-determine-version:

-phar-build:
     [copy] Copying 225 files to /usr/local/src/phpunit/build/phar/phpunit
[php-scoper] PHP Warning:  Cannot declare class JsonException, because the name is already in use in phar:///usr/local/src/phpunit/tools/php-scoper/vendor/symfony/polyfill-php73/Resources/stubs/JsonException.php on line 8
[php-scoper] 
[php-scoper] PHP Stack trace:
[php-scoper] PHP   1. {main}() /usr/local/src/phpunit/tools/php-scoper:0
[php-scoper] PHP   2. require() /usr/local/src/phpunit/tools/php-scoper:19
[php-scoper] PHP   3. require() phar:///usr/local/src/phpunit/tools/php-scoper/bin/php-scoper:33
[php-scoper] PHP   4. class_exists() phar:///usr/local/src/phpunit/tools/php-scoper/vendor/autoload.php:13
[php-scoper] PHP   5. spl_autoload_call() phar:///usr/local/src/phpunit/tools/php-scoper/vendor/autoload.php:13
[php-scoper] PHP   6. Composer\Autoload\ClassLoader->loadClass() phar:///usr/local/src/phpunit/tools/php-scoper/vendor/autoload.php:13
[php-scoper] PHP   7. Composer\Autoload\includeFile() phar:///usr/local/src/phpunit/tools/php-scoper/.box/vendor/composer/ClassLoader.php:322
[php-scoper] PHP   8. include() phar:///usr/local/src/phpunit/tools/php-scoper/.box/vendor/composer/ClassLoader.php:444
[php-scoper] PHP   9. class_alias() phar:///usr/local/src/phpunit/tools/php-scoper/vendor/symfony/polyfill-php73/Resources/stubs/JsonException.php:8
[php-scoper] 
[php-scoper]     ____  __  ______     _____                           
[php-scoper]    / __ \/ / / / __ \   / ___/_________  ____  ___  _____
[php-scoper]   / /_/ / /_/ / /_/ /   \__ \/ ___/ __ \/ __ \/ _ \/ ___/
[php-scoper]  / ____/ __  / ____/   ___/ / /__/ /_/ / /_/ /  __/ /    
[php-scoper] /_/   /_/ /_/_/       /____/\___/\____/ .___/\___/_/     
[php-scoper]                                      /_/
[php-scoper] 
[php-scoper] PHP Scoper version 0.12.99@ 2019-06-09 08:34:32 UTC
[php-scoper] 
[php-scoper]    0/654 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%
[php-scoper]   65/654 [▓▓░░░░░░░░░░░░░░░░░░░░░░░░░░]   9%
[php-scoper]  130/654 [▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░░]  19%
[php-scoper]  195/654 [▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░]  29%
[php-scoper]  260/654 [▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░░░]  39%
[php-scoper]  325/654 [▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░░]  49%
[php-scoper]  390/654 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░]  59%
[php-scoper]  455/654 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░]  69%
[php-scoper]  520/654 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░]  79%
[php-scoper]  585/654 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░]  89%
[php-scoper] 
[php-scoper] 
[php-scoper]  [OK] Successfully prefixed 654 files.                                          
[php-scoper] 
[php-scoper]  // Memory usage: 16.49MB (peak: 24.05MB), time: 8.90s                          
[php-scoper] 
[php-scoper]  650/654 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░]  99%
[php-scoper]  654/654 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%
    [phpab] phpab 1.25.6 - Copyright (C) 2009 - 2019 by Arne Blankerts and Contributors
    [phpab] 
    [phpab] Scanning directory /usr/local/src/phpunit/build/phar-scoped
    [phpab] 
    [phpab] phar archive '/usr/local/src/phpunit/build/phpunit-scoped-library-8.2.1-1-g7bf3fc4ca.phar' generated.
    [phpab] 
     [copy] Copying 1 file to /usr/local/src/phpunit/build
    [phpab] phpab 1.25.6 - Copyright (C) 2009 - 2019 by Arne Blankerts and Contributors
    [phpab] 
    [phpab] Scanning directory /usr/local/src/phpunit/build/phar-scoped
    [phpab] 
    [phpab] phar archive '/usr/local/src/phpunit/build/phpunit-scoped-8.2.1-1-g7bf3fc4ca.phar' generated.
    [phpab] 
   [delete] Deleting directory /usr/local/src/phpunit/build/phar
   [delete] Deleting directory /usr/local/src/phpunit/build/phar-scoped
   [delete] Deleting: /usr/local/src/phpunit/build/binary-phar-autoload.php

signed-scoped-phar:

BUILD SUCCESSFUL
Total time: 17 seconds
```

### PR DESCRIPTION
**Pull request by theofidry**
Closes #336
Closes #338

### CODE REVIEW

**Comment by muglug** [comment]
Does `JsonException` get whitelisted automatically? I’m not whitelisting it explicitly.

**Comment by theofidry** [comment]
No it was actually from PHP-Scoper itself. I didn't notice the issue in the CI because there was only a PHP warning being displayed instead of throwing an error (due to the error handler being registered too late to kick in).

The class itself is whitelisted because it comes from a dedicated file, which means it will be scoped (otherwise the autoloading for that package could be broken) hence the only way to expose the original class was to whitelist it (which was done automatically by PHP-Scoper).

TL:DR the only cleaner solution would be for the polyfill to declare the class _only_ if the class is missing with a `class_exists()`, but I think adding a guard at this level doesn't hurt either and solve that issue...
